.github/workflows/tests.yml

```diff
@@ -51,3 +51,11 @@ jobs:
     - name: Stop containers
       if: always()
       run: scripts/docker down
+    - name: Upload test artifacts
+      if: always()
```

Comment:
Why is this `if` clause necessary? Seems like no.

Reply:
Maybe I'm missing something about how GH Actions `steps` work. Are you saying this step will always run (perhaps the `strategy.fail-fast` ensures this, or is it that the previous job will never fail)?